File: sdk-python/examples/invoices/retrieve.py
"""Run with: python examples/invoices/retrieve.py"""

from __future__ import annotations

from threecommon import ThreeCommon


def main() -> None:
    with ThreeCommon(api_key="3co_your_api_key_here") as client:
        inv = client.invoices.retrieve("inv_replace_with_real_id")
        print(f"invoice {inv.id} [{inv.status}]")
        print(f"  total:       {inv.total} {inv.currency}")
        print(f"  amount_paid: {inv.amount_paid}")
        print(f"  amount_due:  {inv.amount_due}")
        print(f"  line items:  {len(inv.line_items or [])}")
        print(f"  payments:    {len(inv.payments or [])}")


if __name__ == "__main__":
    main()

File: sdk-node/examples/invoices/retrieve.ts
/**
 * Retrieve a single invoice by ID, with line items and payments.
 *
 * Run:
 *   npx tsx examples/invoices/retrieve.ts
 */

import { ThreeCommon } from '@3common/sdk'

const client = new ThreeCommon({
  apiKey: '3co_your_api_key_here',
})

const invoice = await client.invoices.retrieve('inv_replace_with_real_id')

console.log(`invoice ${invoice.id ?? '?'} [${invoice.status ?? '?'}]`)
console.log(`  total      ${String(invoice.total ?? 0)} ${invoice.currency ?? ''}`)
console.log(`  paid       ${String(invoice.amountPaid ?? 0)}`)
console.log(`  due        ${String(invoice.amountDue ?? 0)}`)
console.log(`  line items: ${String(invoice.lineItems?.length ?? 0)}`)
console.log(`  payments:   ${String(invoice.payments?.length ?? 0)}`)
